app/controllers/reviews_controller.rb

```diff
@@ -33,23 +33,27 @@ def edit
   def create
     @review = Review.new(review_params)
 
-    respond_to do |format|
-      if @review.save
-        format.html do
-          redirect_to professor_path(@review.professor_id),
-                      notice: 'Review was successfully created.'
-        end
-        format.json { render :show, status: :created, location: @review }
-      else
-        format.html do
-          @courses = load_courses
-          @professors = load_professors
-          render :new
-        end
-        format.json do
-          render json: @review.errors, status: :unprocessable_entity
+    if (params[:course_id] && params[:course_id] != '') &&
+         (params[:professor_id] && params[:professor_id] != '')
```

Comment:
I _believe_ that this statement can be consolidated a bit using Rails `.blank?` function:

```suggestion
    if params[:course_id].blank? || params[:professor_id].blank?
        redirect_to new_review_path, alert: 'Fill out all required fields'
        return
    end
```

The rest can stay the same (except for the next two statements)